### DISCUSSION TITLE
How long to support Python 3.7, and by extension Python 3.8.

### DISCUSSION THREAD

**Discussion by wynquinox** [grpc-requests/grpc_requests#81] [General]
As the title states.

Python 3.7 went end of life in June of 2023, and 3.8 is going to go end of life in October of this year.

Does anyone have any thoughts on went to stop guaranteeing support for out of support iterations of the language?

Context - if/when we move to some cleanups for 0.2.0 - I'm thinking removing old 3.7 hold overs might be a nice cleanup.

**Comment by aidandj** [comment]
I think the additional ruff abilities by getting rid of 3.7 are worth it. 3.8 on isn't much different in my experience (just did a 3.8 -> 3.12 move)

3.9 has a bunch of nicer syntax

**Reply by wynquinox** [reply]
Wholly agreed there on 3.9.

I'm pretty down with cutting 3.7 loose with a bit of notice, and then saying 3.8 will no longer be officially supported after it goes end of life.

On our end, just turn off the PR tests - and if the libraries work with earlier versions, that's a happy accident for those folks.

**Reply by aidandj** [reply]
Is there some reason to set a marker that says >3.8 so that people will automatically get the old versions ? I think some of the syntax changes will render them non functional right away.

**Reply by wynquinox** [reply]
Fair point - I was primarily thinking calling it out in documentation and policy, but we could leave some recommended usages of environment markers so folks' can set their `requirements.txt` appropriately.
